src/Sylius/Behat/Context/Ui/Admin/ManagingProductsContext.php

```diff
@@ -0,0 +1,372 @@
+     */
+    public function iSpecifyItsPriceTo($price)
+    {
+        $this->createSimpleProductPage->specifyPrice($price);
+    }
+
+    /**
+     * @Given the product :productName should appear in the shop
+     * @Given the product :productName should be in the shop
+     */
+    public function theProductShouldAppearInTheShop($productName)
+    {
+        $this->iWantToBrowseProducts();
+
+        Assert::true(
+            $this->indexPage->isSingleResourceOnPage(['name' => $productName]),
+            sprintf('The product with name %s has not been found.', $productName)
+        );
+    }
+
+    /**
+     * @When I want to browse products
+     */
+    public function iWantToBrowseProducts()
+    {
+        $this->indexPage->open();
+    }
+
+    /**
+     * @Then I should see :numberOfProducts products in the list
+     */
+    public function iShouldSeeProductsInTheList($numberOfProducts)
+    {
+        $foundRows = $this->indexPage->countItems();
+
+        Assert::eq(
+            $numberOfProducts,
+            $foundRows,
+            '%s rows with products should appear on page, %s rows has been found'
+        );
+    }
+
+    /**
+     * @When I delete the :product product
+     * @When I try to delete the :product product
+     */
+    public function iDeleteProduct(ProductInterface $product)
+    {
+        $this->sharedStorage->set('product', $product);
+
+        $this->iWantToBrowseProducts();
+        $this->indexPage->deleteResourceOnPage(['name' => $product->getName()]);
+    }
+
+    /**
+     * @Then /^(this product) should not exist in the product catalog$/
+     */
+    public function productShouldNotExist(ProductInterface $product)
+    {
+        $this->iWantToBrowseProducts();
+
+        Assert::false(
+            $this->indexPage->isSingleResourceOnPage(['code' => $product->getCode()]),
+            sprintf('Product with code %s exists but should not.', $product->getCode())
+        );
+    }
+
+    /**
+     * @Then I should be notified that this product is in use and cannot be deleted
+     */
+    public function iShouldBeNotifiedOfFailure()
+    {
+        $this->notificationChecker->checkNotification(
+            "Cannot delete, the product is in use.",
+            NotificationType::failure()
+        );
+    }
+
+    /**
+     * @Then /^(this product) should still exist in the product catalog$/
+     */
+    public function productShouldExistInTheProductCatalog(ProductInterface $product)
+    {
+        $this->theProductShouldAppearInTheShop($product->getName());
+    }
+
+    /**
+     * @When I want to modify the :product product
+     * @When /^I want to modify (this product)$/
+     */
+    public function iWantToModifyAProduct(ProductInterface $product)
+    {
+        $this->sharedStorage->set('product', $product);
+
+        if (ProductInterface::SIMPLE === $product->getType()) {
+            $this->updateSimpleProductPage->open(['id' => $product->getId()]);
```

Comment:
Missing `return`. Right now when product is simple we will first open  `updateSimpleProductPage` and after that `updateConfigurableProductPage`...